### DISCUSSION TITLE
ITP Cookies instructions

### DISCUSSION THREAD

**Discussion by andthink** [osiset/laravel-shopify#736] [Q&A]
Hi 

I understand this package now handles ITP cookies, and that there is a new ITP middleware, and that frontend part is not included.
But I don't understand how am I supposed to use ITP cookies
i just have to add 
`->middleware(['auth.shopify', 'itp'])`

to my admin routes and then deal with ITP cookies in frontend?

Thanks

PS
Can i benefit from ITP cookies in a standard Multiple page application? Or it's useful only for SPA?

**Comment by gnikyt** [comment]
For SPAs it seems people are utilizing JWT, but yes you can, you would just need it on your main route.

The ITP implementation is basic. You can find more information [here](https://github.com/osiset/laravel-shopify/wiki/ITP) on what it does and tries to do for the customer, with fallback cases.

**Reply by andthink** [reply]
Is ITP cookie supposed to solve admin lost cookie/session? I admit I'm not expert and I'm trying to understand if i can benefit from this.

**Reply by gnikyt** [reply]
Yes. It will try to set a cookie on the full app page (not in iframe), then it checks if it worked. If it did not work, it sends you to a page to try and enable cookies with permission from the browser. If that fails to work, it tells you to manually enable cookies.

[Info here](https://github.com/osiset/laravel-shopify/wiki/ITP)
